src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -1736,8 +1736,9 @@ await PreviewBuildIntegratedProjectActionsAsync(buildIntegratedProject, actions,
                     var primaryPackages = new List<PackageIdentity> { packageIdentity };
 
                     HashSet<SourcePackageDependencyInfo> availablePackageDependencyInfoWithSourceSet = null;
+                    PackageNamespacesConfiguration packageNamespacesConfiguration = PackageNamespacesConfiguration.GetPackageNamespacesConfiguration(Settings);
 
-                    var gatherContext = new GatherContext()
+                    var gatherContext = new GatherContext(packageNamespacesConfiguration)
```

Comment:
I think there's a good chance to add some tests for the NuGetPackageManager class as well.

Reply:
Added.